internal/resources/resource_lb_virtual_servers.go

```diff
@@ -15,87 +17,123 @@ import (
 func LoadBalancerVirtualServers() *schema.Resource {
 	return &schema.Resource{
 		Schema: map[string]*schema.Schema{
-			"vip_name": {
-				Type:        schema.TypeString,
+			"lb_id": {
+				Type:        schema.TypeInt,
 				Required:    true,
+				Description: "Parent lb ID, lb_id can be obtained by using LB datasource/resource.",
+				ForceNew:    true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
 				Description: "vip_name of Network loadbalancer virtual server name",
 			},
 			"description": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "description of Network loadbalancer virtual server",
-				ForceNew:    true,
 			},
 			"vip_address": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "vip_address of Network loadbalancer virtual server",
-				ForceNew:    true,
 			},
 			"vip_port": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				Description: "vip_port of network loadbalancer virtual server",
 			},
 			"pool": {
 				Type:        schema.TypeInt,
-				Required:    true,
+				Optional:    true,
 				Description: "pool of Network loadbalancer virtual server",
 			},
-			"ssl_cert": {
-				Type:        schema.TypeInt,
-				Required:    true,
-				Description: "ssl_cert of Network loadbalancer virtual server",
+			"vip_host_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "vip_host_name of Network loadbalancer virtual server",
+			},
+			"type": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validations.StringInSlice([]string{"http", "tcp", "udp"}, false),
+				Optional:         true,
+				Default:          "http",
+				Description:      "vip protocol of Network loadbalancer virtual server",
 			},
+			"tcp_application_profile":  schemas.TcpAppProfileSchema(),
+			"udp_application_profile":  schemas.UdpAppProfileSchema(),
+			"http_application_profile": schemas.HttpAppProfileSchema(),
+			"persistence": {
+				Type:             schema.TypeString,
+				ValidateDiagFunc: validations.StringInSlice([]string{"SOURCE_IP", "COOKIE"}, false),
+				Optional:         true,
+				Description:      "vip_host_name of Network loadbalancer virtual server",
+			},
+			"cookie_persistence_profile":   schemas.CookiePersProfileSchema(),
+			"sourceip_persistence_profile": schemas.SourceipPersProfileSchema(),
 			"ssl_server_cert": {
 				Type:        schema.TypeInt,
-				Required:    true,
+				Optional:    true,
```

Comment:
why is this Int ?

Reply:
yes the value is int only getting from the datasource